### PR TITLE
Remove App Engine project ID from Maven New Project Wizard

### PR DESCRIPTION
**Pull request by elharo**

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/CreateMavenBasedAppEngineStandardProject.java [hunk 1]
```diff
@@ -84,7 +76,6 @@ protected void execute(IProgressMonitor monitor)
           project, true, loopMonitor.newChild(1));
       AppEngineStandardFacet.installAppEngineFacet(facetedProject, true /* installDependentFacets */, loopMonitor.newChild(1));
       AppEngineStandardFacet.installAllAppEngineRuntimes(facetedProject, true /* force */, loopMonitor.newChild(1));
-      PreferenceUtil.setProjectIdPreference(project, appId);
```
Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/CreateMavenBasedAppEngineStandardProject.java]: We could remove the `PreferenceUtil` class (and corresponding test class) as well, as nothing else uses it.

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java [hunk 2]
```diff
@@ -182,15 +180,6 @@ private void createAppEngineProjectDetailsArea(Composite container) {
     javaPackagePosition.horizontalSpan = 2;
     javaPackageField.setLayoutData(javaPackagePosition);
     javaPackageField.addModifyListener(pageValidator);
-
-    // App Engine Project ID
-    Label projectIdLabel = new Label(container, SWT.NONE);
-    projectIdLabel.setText(Messages.getString("PROJECT_ID_LABEL")); //$NON-NLS-1$
```
Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java]: Can we remove this from messages?

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java [hunk 3]
```diff
@@ -296,11 +285,6 @@ private boolean validateAppEngineProjectDetails() {
       return false;
     }
 
-    String projectId = getAppEngineProjectId();
-    if (!projectId.isEmpty() && !ProjectIdValidator.validate(projectId)) {
-      setErrorMessage(MessageFormat.format(Messages.getString("ILLEGAL_PROJECT_ID"), projectId)); //$NON-NLS-1$
```
Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java]: Can we remove this from messages?

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/CreateMavenBasedAppEngineStandardProject.java (hunk 1 above)
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/CreateMavenBasedAppEngineStandardProject.java]: This turns out to be a completely separate bundle, so let's address that in a new PR.

Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/CreateMavenBasedAppEngineStandardProject.java]: Took another look. This makes sense. Done.

**Comment by elharo** [comment]
Weird, not sure why Travis checks suddenly disappeared from this PR

**Comment by elharo** [comment]
The continued failures in 

com.google.cloud.tools.eclipse.integration.appengine.NewMavenBasedAppEngineProjectWizardTest
testHelloWorldInTemp(com.google.cloud.tools.eclipse.integration.appengine.NewMavenBasedAppEngineProjectWizardTest)  Time elapsed: 44.472 sec
testGuestbookExample(com.google.cloud.tools.eclipse.integration.appengine.NewMavenBasedAppEngineProjectWizardTest)  Time elapsed: 61.423 sec  <<< ERROR!

are bedeviling me. If anyone sees why these might be failing, please chime in.

**Comment by chanseokoh** [comment]
Because of this?

```
!MESSAGE Unable to create project from archetype [com.google.appengine.archetypes:guestbook-archetype:LATEST]
!STACK 0
org.apache.maven.archetype.exception.ArchetypeNotConfigured: Archetype com.google.appengine.archetypes:guestbook-archetype:3.0.2-1.9.38 is not configured
        Property application-id is missing.
```

**Comment by elharo** [comment]
On Mon, Oct 24, 2016 at 10:43 AM, Chanseok Oh <notifications@github.com>
wrote:

> Because of this?
>
> !MESSAGE Unable to create project from archetype [com.google.appengine.archetypes:guestbook-archetype:LATEST]
> !STACK 0
> org.apache.maven.archetype.exception.ArchetypeNotConfigured: Archetype com.google.appengine.archetypes:guestbook-archetype:3.0.2-1.9.38 is not configured
>         Property application-id is missing.
>
>
Makes sense. A second pair of eyes always helps. Let me look at that.


-- 
Elliotte Rusty Harold
elharo@macfaq.com

**Pull request merged**